<a href="https://colab.research.google.com/github/AshleyHeinrich/HealthcareClassification/blob/main/Project_2_Options.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Heart Failure Prediction

In [ ]:
## Pandas
import pandas as pd
## Numpy
import numpy as np

#Dataset
filename = ('https://docs.google.com/spreadsheets/d/e/2PACX-1vSk72yHv2ST8JoGXgy6iGZBVkbm1OvEwOirKBlJJJmHL0hML-lAR5s-Ix7djhxi1ctz3ozkOizryzby/pub?output=csv')
df = pd.read_csv(filename)

Attribute Information

Age: age of the patient [years]

*Sex: sex of the patient [M: Male, F: Female]
*ChestPainType: chest pain type [TA: Typical Angina, ATA: Atypical *Angina, NAP: Non-Anginal Pain, ASY: Asymptomatic]
*RestingBP: resting blood pressure [mm Hg]
*Cholesterol: serum cholesterol [mm/dl]
*FastingBS: fasting blood sugar [1: if FastingBS > 120 mg/dl, 0: otherwise]
*RestingECG: resting electrocardiogram results [Normal: Normal, ST: having ST-T wave abnormality (T wave inversions and/or ST elevation or depression of > 0.05 mV), LVH: showing probable or definite left ventricular hypertrophy by Estes' criteria]
*MaxHR: maximum heart rate achieved [Numeric value between 60 and 202]
*ExerciseAngina: exercise-induced angina [Y: Yes, N: No]
*Oldpeak: oldpeak = ST [Numeric value measured in depression]
*ST_Slope: the slope of the peak exercise ST segment [Up: upsloping, Flat: flat, Down: downsloping]
*HeartDisease: output class [1: heart disease, 0: Normal]

In [ ]:
df.head()

Age Sex ChestPainType  RestingBP  Cholesterol  FastingBS RestingECG  MaxHR  \
0   40   M           ATA        140          289          0     Normal    172   
1   49   F           NAP        160          180          0     Normal    156   
2   37   M           ATA        130          283          0         ST     98   
3   48   F           ASY        138          214          0     Normal    108   
4   54   M           NAP        150          195          0     Normal    122   

  ExerciseAngina  Oldpeak ST_Slope  HeartDisease  
0              N      0.0       Up             0  
1              N      1.0     Flat             1  
2              N      0.0       Up             0  
3              Y      1.5     Flat             1  
4              N      0.0       Up             0

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 918 entries, 0 to 917
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Age             918 non-null    int64  
 1   Sex             918 non-null    object 
 2   ChestPainType   918 non-null    object 
 3   RestingBP       918 non-null    int64  
 4   Cholesterol     918 non-null    int64  
 5   FastingBS       918 non-null    int64  
 6   RestingECG      918 non-null    object 
 7   MaxHR           918 non-null    int64  
 8   ExerciseAngina  918 non-null    object 
 9   Oldpeak         918 non-null    float64
 10  ST_Slope        918 non-null    object 
 11  HeartDisease    918 non-null    int64  
dtypes: float64(1), int64(6), object(5)
memory usage: 86.2+ KB


In [ ]:
df.duplicated().sum()

0

In [ ]:
df['HeartDisease'].value_counts()
#imbalanced data

1    508
0    410
Name: HeartDisease, dtype: int64

1. This dataset has 11 features, many of which could be used to predict possible heart disease. Information like cholesterol, blood pressure, fasting blood sugar, etc and results for each patient/individual.  
2. The target would be Heart Disease:1 or No Disease:0
3. Each row represents an individual 
4. This is a classification problem
5. There are 918 rows and 12 columns 
6. There isn't any missing data or duplicated rows to contend with. While it appears to be clean and complete the challenges will be reducing false negatives and getting the highest accuracy and recall scores possible while not overly compromising precision. This will include model selection, tuning thats impactful and generates significant positive changes, etc. 



# Metabolic Syndrome 
(preferred)

In [ ]:
filename=('https://docs.google.com/spreadsheets/d/e/2PACX-1vRZCC3KLSaYXzE69EA2IYa6udd-0VbkqlhrT08i0OUD1JX8sU5uIG4IVG5wlvQ8SZnfyKz3FnTuGcux/pub?output=csv')
df_2 = pd.read_csv(filename)
df_2.head()

seqn  Age     Sex  Marital  Income   Race  WaistCirc   BMI  Albuminuria  \
0  62161   22    Male   Single  8200.0  White       81.0  23.3            0   
1  62164   44  Female  Married  4500.0  White       80.1  23.2            0   
2  62169   21    Male   Single   800.0  Asian       69.6  20.1            0   
3  62172   43  Female   Single  2000.0  Black      120.4  33.3            0   
4  62177   51    Male  Married     NaN  Asian       81.1  20.1            0   

   UrAlbCr  UricAcid  BloodGlucose  HDL  Triglycerides MetabolicSyndrome  
0     3.88       4.9            92   41             84         No MetSyn  
1     8.55       4.5            82   28             56         No MetSyn  
2     5.07       5.4           107   43             78         No MetSyn  
3     5.22       5.0           104   73            141         No MetSyn  
4     8.13       5.0            95   43            126         No MetSyn

In [ ]:
df_2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2401 entries, 0 to 2400
Data columns (total 15 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   seqn               2401 non-null   int64  
 1   Age                2401 non-null   int64  
 2   Sex                2401 non-null   object 
 3   Marital            2193 non-null   object 
 4   Income             2284 non-null   float64
 5   Race               2401 non-null   object 
 6   WaistCirc          2316 non-null   float64
 7   BMI                2375 non-null   float64
 8   Albuminuria        2401 non-null   int64  
 9   UrAlbCr            2401 non-null   float64
 10  UricAcid           2401 non-null   float64
 11  BloodGlucose       2401 non-null   int64  
 12  HDL                2401 non-null   int64  
 13  Triglycerides      2401 non-null   int64  
 14  MetabolicSyndrome  2401 non-null   object 
dtypes: float64(5), int64(6), object(4)
memory usage: 281.5+ KB


In [ ]:
df_2.duplicated().sum()

0

In [ ]:
df_2['seqn'].value_counts()

62161    1
68696    1
68702    1
68703    1
68706    1
        ..
65410    1
65412    1
65416    1
65421    1
71915    1
Name: seqn, Length: 2401, dtype: int64

In [ ]:
df_2['MetabolicSyndrome'].value_counts()
#imbalanced

No MetSyn    1579
MetSyn        822
Name: MetabolicSyndrome, dtype: int64

In [ ]:
df_2.isna().sum()

seqn                   0
Age                    0
Sex                    0
Marital              208
Income               117
Race                   0
WaistCirc             85
BMI                   26
Albuminuria            0
UrAlbCr                0
UricAcid               0
BloodGlucose           0
HDL                    0
Triglycerides          0
MetabolicSyndrome      0
dtype: int64

In [ ]:
df_2.isna().sum() / len(df_2) * 100

seqn                 0.000000
Age                  0.000000
Sex                  0.000000
Marital              8.663057
Income               4.872970
Race                 0.000000
WaistCirc            3.540192
BMI                  1.082882
Albuminuria          0.000000
UrAlbCr              0.000000
UricAcid             0.000000
BloodGlucose         0.000000
HDL                  0.000000
Triglycerides        0.000000
MetabolicSyndrome    0.000000
dtype: float64

1. This dataset has 2401 rows and 15 colums
2. There are 14 features. Will drop the seqn column because of high cardinality. Dataset includes features/characteristics that can be used to define Metabolic Syndrome (blood sugar, waist circ, etc),along with others that may or may not correlate with the disease. 
3. This is a classification problem to predict the development of metabolic syndrome. 
4. The target is Metabolic Syndrome: 1 or No Metabilic Syndrom: 0 
5. Each row represents an individual
6. Challenges in data cleaning include dealing with missing data and whether to impute, delete or some combination of these two. 
7. Background information like a data dictionary for this dataset is missing regarding how they defined metabolic syndrome (NIH ATP III definiton or WHO as an example differ) or just to provide information about columns. There is also no units related to data points so assumptions will have to be made regarding these unless they can be found. 

Link has a table that summarizes different ways metabolic syndrome is defined. 
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2675814/


Extra: After graduation, I will most likely will be targeting healthcare related data analyst positions. Having a project with relevance to my field would be really helpful. 